eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -96,7 +97,34 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, router: ActorRef
       require(remoteNodeId == remoteNodeId1, s"invalid nodeid: $remoteNodeId != $remoteNodeId1")
       log.debug(s"got authenticated connection to $remoteNodeId@${address.getHostString}:${address.getPort}")
 
-      context watch peerConnection
+      // We want to log all incoming/outgoing messages to/from this peer. Logging incoming messages is easy (they all go
+      // through this actor), but for outgoing messages it's a bit trickier because channels directly send messages to
+      // the connection. That's why we use this pass-through actor that just logs messages and forward them.
+      val logConnection = context.actorOf(Props(new Actor {
```

Comment:
It feels like we could generalize this. Maybe creating a `LightningMessageLogger` actor that wraps a child actor and takes some extra configuration for some MDC values that should be fixed (`remoteNodeId` in this case)?

It feels like `Peer` could also be wrapped in such a logging actor instead of defining almost the same thing inside an `aroundReceive`.

Not entirely sure it would work, just my 2 cents.

Reply:
The problem we have here is that incoming messages are received by the `Peer`, but outgoing messages (mostly sent by `Channel`) are directly sent to the `PeerConnection`, thus bypassing the `Peer`. I don't see how your proposal addressed this issue.

Reply:
Yes but that's completely symmetrical: the code for `logMsgIn` here is very close to the code for `logMsgOut`.
It feels to me that this `logMsgOut` is just the same of `logMsgIn`, but from the perspective of the `PeerConnection`, isn't it?

Reply:
There should be an explicit comment that explains that `logConnection` is to be used as `remote` whenever we send a `connected` notification.

Reply:
Tried a different strategy with 647461f, wdyt?

Reply:
I like https://github.com/ACINQ/eclair/commit/647461f9edc3b2b75bb61f597cfa3533a1024e12 better than the logging actor, I think it's clearer. I don't think the forwarder was really needed, the changes in `Channel.scala` aren't very scary. It's a bit easier to follow the flow if we know that everything goes through the peer IMO.

Maybe we can use an explicit type instead of a `Tuple` in the channel's `send`, and a couple refactoring in that commit and it sounds good to me.

Reply:
Okay, let's see what @sstone thinks.

It does feel better, even if it messes up the tests a little bit.

Reply:
It looks much better indeed but I had the same reaction as @t-bast : using an Option[ActorRef ] to store the channel's active connection feels a bit more natural

Reply:
Wdyt of my answer?

> I think deadletters is better here, because this is what will happen in the normal case when a connection dies. Allows us to only have 2 different states.